app/models/grda_warehouse/youth/zip_exporter.rb

```diff
@@ -0,0 +1,71 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'zip'
+module GrdaWarehouse::Youth
+  class ZipExporter
+    def initialize(intakes:, referrals:, dfas:, case_managements:, follow_ups:, housing_resolution_plans:, controller:, file_path: 'var/yya_export')
+      @intakes = intakes
+      @referrals = referrals
+      @dfas = dfas
+      @case_managements = case_managements
+      @follow_ups = follow_ups
+      @housing_resolution_plans = housing_resolution_plans
+      @controller = controller
+      @file_path = "#{file_path}/#{Process.pid}" # Usual Unixism -- create a unique path based on the PID
+    end
+
+    def export!
+      create_export_directory
+      begin
+        @intakes.each do |intake|
+          locals = {
+            client_intake: intake,
+            client_referrals: @referrals.select { |referral| referral.client_id == intake.client_id },
+            client_dfas: @dfas.select { |dfa| dfa.client_id == intake.client_id },
+            client_case_managements: @case_managements.select { |case_management| case_management.client_id == intake.client_id },
+            client_follow_ups: @follow_ups.select { |follow_up| follow_up.client_id == intake.client_id },
+            housing_resolution_plans: @housing_resolution_plans.select { |hrp| hrp.client_id == intake.client_id },
+          }
+          contents = @controller.render_to_string(:per_client, locals: locals)
+          filename = File.join(@file_path, 'Client ' + intake.client_id.to_s + (intake.hmis_client? ? ' (HMIS)' : '') + '.xlsx')
```

Comment:
I believe a client might have multiple intakes during the date range.  These file names should probably be: 
```suggestion
          name = "Client #{intake.client_id} - #{intake.id}"
          name += ' (HMIS)' if intake.hmis_client?
          name += '.xlsx'
          filename = File.join(@file_path, name)
```

Reply:
or alternatively, group all intakes by client_id first and then add multiple to the first sheet, but I think I like one per file better.